docs(change-logs): announce upcoming removal of LESS files (#4868)
SoftwareAG/c8y-docs: content/change-logs/application-enablement/ui-c8y-removal-of-less-files-announcement.md

---
date: '2026-07-01'
title: Upcoming removal of LESS files following the migration to SCSS
product_area: Application enablement & solutions
change_type:
  - value: change-inv-3bw8e
    label: Announcement
component:
  - value: component-YbYJ3gLU_
    label: Web SDK
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-67215
environment_availability:
---

> The {{< product-c8y-iot >}} Web SDK styles in the `@c8y/style` package have been migrated from LESS to SCSS. As a follow-up, the redundant `.less` files will be **removed** from `@c8y/style`. This change is currently in develop and will ship in an upcoming Continuous Deployment (CD) version of the 2026 release; a separate announcement will confirm the exact version.

Reviewer: ```suggestion
The {{< product-c8y-iot >}} Web SDK styles in the `@c8y/style` package have been migrated from LESS to SCSS. As a follow-up, the redundant `.less` files will be **removed** from `@c8y/style`. This change is currently available in the `develop` branch and will ship in an upcoming Continuous Deployment (CD) version of the 2026 release. The exact release version will be announced separately. 
```

> The {{< product-c8y-iot >}} Web SDK styles in the `@c8y/style` package have been migrated from LESS to SCSS. As a follow-up, the redundant `.less` files will be **removed** from `@c8y/style`. This change is currently on the `develop` branch and will ship in an upcoming Continuous Deployment (CD) version of the 2026 release; a separate announcement will confirm the exact version.

Reviewer 2: ```suggestion
The {{< product-c8y-iot >}} Web SDK styles in the `@c8y/style` package have been migrated from LESS to SCSS. As a follow-up, the redundant `.less` files will be **removed** from `@c8y/style`. This change is currently in develop and will ship in an upcoming Continuous Deployment (CD) version of the 2026 release; a separate announcement will confirm the exact version.
```
I would remove the internal details.

> The {{< product-c8y-iot >}} Web SDK styles in the `@c8y/style` package have been migrated from LESS to SCSS. As a follow-up, the redundant `.less` files will be **removed** from `@c8y/style`. This change is currently available in the `develop` branch and will ship in an upcoming Continuous Deployment (CD) version of the 2026 release. The exact release version will be announced separately. 

Reviewer 3: @rahulc8y @janhommes 
"This change is currently available in the `develop` branch and will ship in an upcoming Continuous Deployment (CD) version of the 2026 release."
This sentence sounds weird to me. Shouldn´t it be:
"This change is currently available in the Continuous Deployment (CD) version and will ship in an upcoming maintenance version of the 2026 release."

Reviewer: the comment seems valid to me

Reviewer 3: I will take care and change it.

**What changes?**

All styling is now authored in SCSS. The `.less` files, LESS variables, and LESS mixins previously shipped in `@c8y/style` are being removed, and the package entry point moves from `main.less` to `main.scss`. The SCSS files provide the same variables, mixins, and compiled output.

**Breaking change**

> Custom applications, plugins, or themes that import the removed `.less` files — or that override LESS variables and mixins — will no longer build after upgrading. The build fails with a `Module not found` error for the removed paths.

Reviewer: ```suggestion
Custom applications, plugins, or themes that import the removed `.less` files — or that override LESS variables and mixins — will no longer build after upgrading. The build will fail with a `Module not found` error for the removed paths.
```

**Impact**: Any UI customization that relies on the LESS styles is affected, including:

* Direct imports of `@c8y/style` `.less` files, for example `@c8y/style/main.less`, `@c8y/style/branding.less`, or `@c8y/style/login.less`.
* Overrides of LESS variables or mixins imported from `@c8y/style/variables/index.less` and the related `_brand-vars` and `_color-vars` files.
* Custom themes and branding built on the LESS variables.

Applications and plugins that already consume the SCSS styles are not affected.

**Migration**: Switch your imports and overrides to the SCSS equivalents:

* Replace each `.less` import with the matching `.scss` file, for example `@c8y/style/main.scss` and `@c8y/style/branding.scss`.
* Import variables from `@c8y/style/variables/index.scss` instead of the `.less` file.
* Update the import syntax from LESS `@import 'file.less'` to SCSS `@use 'file'` or `@forward 'file'`, and change LESS variable overrides from `@variable-name` to `$variable-name`.

> For the full list of supported SCSS variables and mixins, refer to the styling documentation. <!-- TODO: add SCSS styling / Codex migration URL once available -->

Reviewer 2: Is this TODO intended?

Author: yes, working on it
